File: backend/api/get-categories.ts
import type { VercelRequest, VercelResponse } from '@vercel/node';

export default async function handler(
  req: VercelRequest,
  res: VercelResponse,
) {
  // 環境変数からIDを安全に取得
  const appId = process.env.RAKUTEN_APP_ID;

  if (!appId) {
    return res.status(500).json({ error: 'API認証情報がサーバー側で設定されていません。' });
  }

  const apiUrl = `https://app.rakuten.co.jp/services/api/Recipe/CategoryList/20170426?format=json&applicationId=${appId}`;

  try {
    const apiResponse = await fetch(apiUrl);
    const data = await apiResponse.json();

    if (apiResponse.ok) {
      // 必要なカテゴリ情報だけを整形して返す
      const categories = {
        large: data.result.large,
        medium: data.result.medium,
      };
      res.status(200).json(categories);
    } else {
      res.status(apiResponse.status).json(data);
    }
  } catch (error) {
    res.status(500).json({ error: 'サーバーでエラーが発生しました。' });
  }
}

File: backend/api/get_categories.py
import os
import httpx
from fastapi import APIRouter, HTTPException

router = APIRouter()

@router.get("/categories")  # ✅ REST風にURL変更
async def get_categories():
    try:
        app_id = os.getenv("RAKUTEN_APP_ID")
        if not app_id:
            raise RuntimeError("環境変数 RAKUTEN_APP_ID が設定されていません。")

        url = f"https://app.rakuten.co.jp/services/api/Recipe/CategoryList/20170426?format=json&applicationId={app_id}"

        async with httpx.AsyncClient() as client:
            response = await client.get(url)
            response.raise_for_status()
            data = response.json()

        return {
            "large": data["result"]["large"],
            "medium": data["result"]["medium"]
        }

    except Exception as e:
        print(f"❌ get_categories() 例外発生: {e}")
        raise HTTPException(status_code=500, detail=str(e))
